### DISCUSSION TITLE
Allow cache for deployment_status

### DISCUSSION THREAD

**Discussion by enricoschaaf** [community/community#160598] [Actions]
Currently the caching if disabled for deplyment_status events with the following warning.
Warning: Event Validation Error: The event type deployment_status is not supported because it's not tied to a branch or tag ref.
Would it be possible to allow this event because in some cases it should be possible to add a ref.

I run checkout with fetch-depth: 0 and get the ref like this
```bash
ref=$(git name-rev --name-only $GITHUB_SHA)
echo refs/heads/${ref##*/}
```

**Comment by hensansi** [comment]
Linking the discussion with an issue,
https://github.com/actions/cache/issues/598

Hopefully, there will be enough asking from the community to support `deployment_status` events. 


From the issue [598](https://github.com/actions/cache/issues/598#issuecomment-1128622845):
_Caches cannot be created without GITHUB_REF as per the current design because the scope is one of the mandatory properties of a cache. Hence it is not possible to take this up at the current time. **We can revisit if there is more ask from the community.**_

**Reply by saiichihashimoto** [reply]
Not sure if this is the right place, but adding to the ask from the community. I have one step in my workflow caching my `npm ci --ignore-scripts` to be used in my other steps and it was an enormous time saving until I needed playwright tests against my deployment. Now that step doubled my deployment time, most of it taken up by having to run it's own `npm ci --ignore-scripts`.
